cmd/secrets-provider/main.go

```diff
@@ -85,7 +91,7 @@ func main() {
 				return log.RecordedError(messages.CSPFK003E, err.Error())
 			}
 
-			if authnConfig.ContainerMode == "init" {
+			if authnConfig.ContainerMode == "init" || envValMultiApps == "true" {
```

Comment:
so `SUPPORT_MULTIPLE_APPS` indicates that the secrets-provider is running in its own pod?

Reply:
here's an idea. the `CONTAINER_MODE` variable can get 3 values:
- `init`
- `sidecar`
- `app`

this still makes sense in our purpose as what you want to say is "if the secrets-provider runs as an init container or if it runs as an app container, we should exit the code after one cycle". this way the only code change will be:
```
if authnConfig.ContainerMode == "init" || authnConfig.ContainerMode == "app" {
  os.Exit(0)
}
```

no extra variable that is not really needed and we get a logical solution to our issue.

what say you?

Reply:
From k8s docs:
```
As well as application containers, a Pod can contain init containers that run during Pod startup
```
so we have the following container types:
- application
- init
- sidecar

it all fits

Reply:
I think it makes it much more clear! Thanks for the input! <3